Core/Core/SwiftUIViews/CoreDatePicker.swift

```diff
@@ -0,0 +1,139 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2020-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+public struct CoreDatePicker {
+
+    public static func pickDate(for date: Binding<Date?>, with dateRange: ClosedRange<Date>? = nil, from controller: UIViewController) {
+        let env = AppEnvironment.shared
+        let dateRange = dateRange ?? Clock.now.addYears(-1)...Clock.now.addYears(1)
+        let picker = CoreHostingController(CoreDatePickerActionSheetCard(selection: date, dateRange: dateRange))
+        picker.view.backgroundColor = UIColor.clear
+        env.router.show(picker,
+                        from: controller,
+                        options: .modal(.overFullScreen,
+                                        isDismissable: true,
+                                        embedInNav: false,
+                                        addDoneButton: false))
+    }
+
+    public static func pickDate(for date: Binding<Date?>, with dateRange: ClosedRange<Date>? = nil, from controller: WeakViewController) {
+        self.pickDate(for: date, with: dateRange, from: controller.value)
+    }
+}
+
+public struct CoreDatePickerActionSheetCard: View {
+    @Environment(\.viewController) var controller
+    @State private var grayViewOpacity = 0.0
+    @State private var animationOffset = 300.0
+    @State private var offset = UIScreen.main.bounds.height
+    @State private var selectedDate: Date
+    @Binding private var selectionDate: Date?
+
+    private let heightToDisappear = UIScreen.main.bounds.height
+    private let backgroundColor = Color.backgroundLightest
+
+    private var pickerDateRange: ClosedRange<Date>
+
+    public init(selection: Binding<Date?>, dateRange: ClosedRange<Date> = Clock.now...Clock.now.addYears(1)) {
+        _selectionDate = selection
+        _selectedDate = State<Date>(initialValue: selection.wrappedValue ?? Clock.now)
+        pickerDateRange = dateRange
+    }
+
+    public var body: some View {
+        ZStack {
+            grayView
+                .opacity(grayViewOpacity)
+                .animation(.easeInOut.delay(0.2), value: grayViewOpacity)
```

Comment:
Somehow the presentation seems to be slow for me. When I tap to show or dismiss the date picker there's a perceptible delay of the show/dismiss animation. Could we improve this somehow maybe?